andela/ah-scorpion-frontend
# #160614287 Allow users to view all articles on the Dashboard page

#### What does this PR do?
Allows users to view all articles on the dashboard page.

#### Description of Task to be completed?

- Create the display component for rendering the articles.
- Re-use the error handling and loading components to give the users visual feedback on the status of the request.
- Wire the container components to Redux.
    - Write the reducer code.
    - Write the actions and action types.
- Add code to make the API calls in `services/apiService.js`
- Add snapshot tests for the display components.
- Add tests for the actions and the reducer.

#### How should this be manually tested?

- Clone the project and checkout to the `ft-view-all-articles-160614287` branch
    - git clone https://github.com/andela/ah-scorpion-frontend.git
    - git checkout `ft-view-all-articles-160614287`
- Copy the variables from `sample.env` and put the correct parameters in `.env` file and initialize them.
    - `source .env`
- Install the project's dependencies.
    - `npm install`
- Start the project
    - `npm start`
- Go to the browser and log in with your account.
- Navigate to `/dashboard`. You can see all Articles in the body of the page.

#### Any background context you want to provide?

- If the user does not provide any images, a blue background is displayed by default.
- The first image provided by the user is automatically picked and used in the card.

#### What are the relevant pivotal tracker stories?

[#160614287](https://www.pivotaltracker.com/story/show/160614287)
#### Screenshots
1. When the page first loads, a request is made to the backend to fetch the articles. This is the loading screen…

![image](https://user-images.githubusercontent.com/28927645/45824728-585a6d80-bcf9-11e8-88a9-1c970d301228.png)


2. If an error occurs during the fetch, this error message is shown.

![image](https://user-images.githubusercontent.com/28927645/45824938-bf782200-bcf9-11e8-88e7-9d9e480720e0.png)


3. If there are no articles, this message is displayed.

![image](https://user-images.githubusercontent.com/28927645/45825061-0534ea80-bcfa-11e8-863c-b2f5d60b07fa.png)


4. When the articles are fetched, they are displayed like so.

![image](https://user-images.githubusercontent.com/28927645/45825120-2564a980-bcfa-11e8-8a05-e78c949b6bba.png)
